### PR TITLE
Implement _mm_hsub_ps

### PR DESCRIPTION
**Pull request by HowJMay**

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -2981,6 +2981,16 @@ FORCE_INLINE __m128i _mm_hadd_epi16(__m128i _a, __m128i _b)
...
+// Horizontally substract adjacent pairs of single-precision (32-bit)
+// floating-point elements in a and b, and pack the results in dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_hsub_ps
+FORCE_INLINE __m128 _mm_hsub_ps(__m128 _a, __m128 _b)
+{
+    float32x4x2_t rev_vec = vuzpq_f32(_a, _b);
+    return vreinterpretq_m128_f32(vsubq_f32(rev_vec.val[0], rev_vec.val[1]));
+}
+
+
```
Comment (jserv) [inline sse2neon.h:2993]: Keep single line separation.

File: sse2neon.h [hunk 2]
```diff
@@ -2981,6 +2981,16 @@ FORCE_INLINE __m128i _mm_hadd_epi16(__m128i _a, __m128i _b)
 #endif
 }
 
+// Horizontally substract adjacent pairs of single-precision (32-bit)
+// floating-point elements in a and b, and pack the results in dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_hsub_ps
+FORCE_INLINE __m128 _mm_hsub_ps(__m128 _a, __m128 _b)
+{
+    float32x4x2_t rev_vec = vuzpq_f32(_a, _b);
```
Comment (jserv) [inline sse2neon.h:2989]: You can provide A64 shortcut as following:
```c
    return vsubq_f32(vuzp1q_f32(_a, _b), vuzp2q_f32(_a,  _b));
```

Check [Neon Intrinsics Reference](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics) for details.

**Review by jserv — CHANGES_REQUESTED** [review]
Ensure A64 path is implemented.

File: sse2neon.h [hunk 3]
```diff
@@ -2981,6 +2981,19 @@ FORCE_INLINE __m128i _mm_hadd_epi16(__m128i _a, __m128i _b)
...
+// Horizontally substract adjacent pairs of single-precision (32-bit)
+// floating-point elements in a and b, and pack the results in dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_hsub_ps
+FORCE_INLINE __m128 _mm_hsub_ps(__m128 _a, __m128 _b)
+{
+#if defined(__aarch64__)
+    float32x4x2_t rev_vec = vuzpq_f32(_a, _b);
+    return vreinterpretq_m128_f32(vsubq_f32(rev_vec.val[0], rev_vec.val[1]));
+#else
+    return vsubq_f32(vuzp1q_f32(_a, _b), vuzp2q_f32(_a, _b));
```
Comment (jserv) [inline sse2neon.h:2993]: Move A64-specific implementation in the top-half of function body.

**Review by jserv — APPROVED** [review]

File: tests/impl.h [hunk 4]
```diff
@@ -218,6 +218,7 @@
     TYPE(mm_hadd_epi16)               \
     TYPE(mm_hadd_epi32)               \
     TYPE(mm_hadds_epi16)              \
+    TYPE(mm_hsub_ps)                  \
```
Comment (marktwtn) [inline tests/impl.h:221]: The class is `SSE3`.

File: sse2neon.h [hunk 5]
```diff
@@ -2981,6 +2981,19 @@ FORCE_INLINE __m128i _mm_hadd_epi16(__m128i _a, __m128i _b)
 #endif
 }
 
+// Horizontally substract adjacent pairs of single-precision (32-bit)
+// floating-point elements in a and b, and pack the results in dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_hsub_ps
+FORCE_INLINE __m128 _mm_hsub_ps(__m128 _a, __m128 _b)
+{
+#if defined(__aarch64__)
+    return vsubq_f32(vuzp1q_f32(_a, _b), vuzp2q_f32(_a, _b));
```
Comment (marktwtn) [inline sse2neon.h:2990]: Add `vreinterpretq_m128_f32` and `vreinterpretq_f32_m128`.

File: sse2neon.h [hunk 6]
```diff
@@ -2981,6 +2981,19 @@ FORCE_INLINE __m128i _mm_hadd_epi16(__m128i _a, __m128i _b)
...
+#else
+    float32x4x2_t rev_vec = vuzpq_f32(_a, _b);
```
Comment (marktwtn) [inline sse2neon.h:2992]: Add `vreinterpretq_f32_m128`.

**Review by marktwtn — CHANGES_REQUESTED** [review]

**Review by marktwtn — APPROVED** [review]

**Pull request merged**